builder/frameworks/espidf.py

```
@@ -195,177 +213,202 @@ def _get_installed_standard_pip_packages():
+                url = file_entry.split(" ")[0]
+                # Path Traversal protection
+                if contains_path_traversal(url):
+                    print(f"Path Traversal detected: {url} check your URL path")
                 else:
-                    print("Failed to download:", file)
-                    return ""
-                return target
-            if "file://" in file:
-                file_path = join(PROJECT_DIR,file.lstrip("file://").split(os.path.sep)[-1])
+                    try:
+                        response = requests.get(file_entry.split(" ")[0])
```

Pylint (3.3.7)

[warning] Missing timeout argument for method 'requests.get' can cause your program to hang indefinitely

(W3101)